src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -259,30 +227,21 @@ class OrderedRVD private(
     if (nPartitions <= 1)
       return filter(pred)
 
+    val rangeTree = partitioner.rangeTree
+
     val intervalArray = intervals.toArray
-    val rangeBounds = partitioner.rangeBounds
-    val partitionIndices = new ArrayBuilder[Int]()
-    var i = 0
-    while (i < nPartitions) {
-      val include = if (i == 0)
-        intervalArray.exists(i => pkOrdering.lteq(i._1.start, rangeBounds(0)))
-      else if (i == nPartitions - 1)
-        intervalArray.reverseIterator.exists(i => pkOrdering.gt(i._1.end, rangeBounds.last))
+    val newPartitionIndices = intervalArray.flatMap { case (i, _) =>
```

Comment:
This should just be:

```
  val newPartitionIndices = intervals.iterator.flatMap { interval =>
      rangeTree.queryValues(interval)
    }
    .toSet // distinct
    .toArray.sorted[Int]
```

Reply:
And then I think you can get rid of `intervalArray`.

Reply:
queryValues only takes positions, not intervals? But I can get rid of intervalArray anyways by using `intervals.toIterable` instead?

Reply:
and I need to get rid of the assert after that because now you can have intervalArray with intervals that don't overlap partition bounds, so `newPartitionIndices` can be empty without `intervals` being empty. (right?)

Reply:
Oops!  I thought we had interval search on interval tree.  We should add that at some point but let's leave the code as-is for now.  (Let me read over it again, I may have alternate suggestions.)

Reply:
Correct (assert is now wrong).  Good catch.